### PR TITLE
Add Java Cloud Run

### PR DESCRIPTION
**Pull request by averikitsch**
Mirroring PR #217 for Java

### CODE REVIEW

**Comment by averikitsch** [comment]
@simonz130 Is there a reason why the tests are not running?

File: .templates/template_location.json [hunk 1]
```diff
@@ -62,6 +62,13 @@
     "templateName": "Java: Hello World",
     "templateLanguages": ["java"]
   },
+  {
+    "repoPath": "https://github.com/GoogleCloudPlatform/cloud-code-samples.git",
+    "directoryPath": "java/java-cloud-run-hello-world",
+    "templatePath": "java-cloud-run-hello-world",
+    "templateName": "Java: Cloud Run Hello World",
+    "templateLanguages": ["java"]
```
Comment (simonz130) [inline .templates/template_location.json]: There should be additional field here:

```json
"runPlatforms": ["cloudrun"]
```

File: cloudbuild.meta.yaml [hunk 2]
```diff
@@ -18,9 +18,9 @@ steps:
   - '-c'
   - |
     lang=java
-    declare -a apps=("hello-world" "guestbook")
-    #gcloud builds submit . --config cloudbuild.tmpl.yaml --substitutions _DIR=${lang},_LANG=${lang},_APP=${apps[0]}
-    #gcloud builds submit . --config cloudbuild.tmpl.yaml --substitutions _DIR=${lang},_LANG=${lang},_APP=${apps[1]}
+    declare -a apps=("hello-world" "guestbook" "cloud-run-hello-world")
```
Comment (simonz130) [inline cloudbuild.meta.yaml]: remove this file - we'll be moving to separate cloud build yaml files per template

File: java/java-cloud-run-hello-world/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+RUN mvn -Dmaven.test.skip=true package
+
+# Build runtime image.
+FROM openjdk:8-jre-alpine
+
+# Copy the compiled files over.
+COPY --from=build-env /app/target/ /app/
+
+# Starts java app with debugging server at port 5005.
+CMD ["java", "-jar", "/app/hello-world-1.0.0.jar"]
```
Comment (simonz130) [inline java/java-cloud-run-hello-world/Dockerfile]: Is the comment correct here? It looks like it starts the app, not debugging server

File: java/java-cloud-run-hello-world/src/main/java/cloudcode/helloworld/web/HelloWorldController.java [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+            .build();
+
+    String metadataUrl = "http://metadata.google.internal/computeMetadata/v1/project/project-id";
+    Request request =
+        new Request.Builder().url(metadataUrl).addHeader("Metadata-Flavor", "Google").get().build();
+
+    String project = null;
+    try {
+      Response response = ok.newCall(request).execute();
+      project = response.body().string();
```
Comment (simonz130) [inline java/java-cloud-run-hello-world/src/main/java/cloudcode/helloworld/web/HelloWorldController.java]: please document this so folks not familiar with metadata service understand that this is the way to retrieve gcp project

File: java/java-cloud-run-hello-world/src/test/java/cloudcode/helloworld/web/HelloWorldControllerTests.java [hunk 5]
```diff
@@ -0,0 +1,31 @@
+package cloudcode.helloworld.web;
+
```
Comment (simonz130) [inline java/java-cloud-run-hello-world/src/test/java/cloudcode/helloworld/web/HelloWorldControllerTests.java]: This is awesome test but I didn't see where you actually invoke it as part of CI/CD pipeline.

File: java/java-cloud-run-hello-world/cloudbuild.yaml [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+- id: 'Integration Tests'
+  name: 'gcr.io/cloud-builders/curl:latest'
+  entrypoint: '/bin/bash'
+  args:
+  - '-c'
+  - |
+    set -ex
+    keyword='Congratulations, you successfully deployed a container image to Cloud Run'
+    chmod +x test_content.sh
+    ./test_content.sh -r 25 -i 3 -u $(cat _service_url) -k $keyword
```
Comment (simonz130) [inline java/java-cloud-run-hello-world/cloudbuild.yaml]: Where is test_content.sh file? I assume that the intention was to wrap invocation of the tests in this file but didn't find it as part of this change.

**Pull request merged**